eessi/reframe/eessi_utils/hooks.py

```diff
@@ -77,6 +87,16 @@ def assign_one_task_per_gpu(test: rfm.RegressionTest) -> rfm.RegressionTest:
     or nb of GPUs per node available in this partition (whatever is smallest).
     """
     max_gpus_per_node = utils.get_num_gpus_per_node(test)
+
```

Comment:
Can you put a comment what this new code section does / is supposed to do? (I think we should probably comment more extensively anyway, but this is a good start :P)

Reply:
while adding the comments, i also made some variable names more clear and self-explanatory, so hopefully it is now easier to understand